baseband/vlbi_base/base.py

```diff
@@ -167,6 +158,30 @@ def thread_ids(self):
     def thread_ids(self, ids):
         self._thread_ids = ids
 
+    @property
+    def samples_per_frame(self):
+        """Number of complete samples per frame."""
+        return self._samples_per_frame
+
+    @samples_per_frame.setter
```

Comment:
Since the setter does not do anything, I would for now remove it, and just set `self._samples_per_frame` in `__init__`. That means you also do not have to set it to `None` in the class.

Alternatively, use a try/except around `self._samples_per_frame = samples_per_frame.__index__()`, i.e., check that it is (convertable to) a proper integer.

Reply:
I like the latter - implementing.